src/Microsoft.ML.Recommender/RecommenderCatalog.cs

```diff
@@ -98,42 +98,42 @@ internal RecommendationTrainers(RecommendationCatalog catalog)
-        /// and respecting <paramref name="stratificationColumn"/> if provided.
-        /// Then evaluate each sub-model against <paramref name="labelColumn"/> and return metrics.
+        /// Run cross-validation over <paramref name="numberOfFolds"/> folds of <paramref name="data"/>, by fitting <paramref name="estimator"/>,
+        /// and respecting <paramref name="samplingKeyColumn"/> if provided.
+        /// Then evaluate each sub-model against <paramref name="labelColumnName"/> and return metrics.
         /// </summary>
         /// <param name="data">The data to run cross-validation on.</param>
         /// <param name="estimator">The estimator to fit.</param>
-        /// <param name="numFolds">Number of cross-validation folds.</param>
-        /// <param name="labelColumn">The label column (for evaluation).</param>
-        /// <param name="stratificationColumn">Optional name of the column to use as a stratification column. If two examples share the same value of the <paramref name="stratificationColumn"/>
+        /// <param name="numberOfFolds">Number of cross-validation folds.</param>
+        /// <param name="labelColumnName">The label column (for evaluation).</param>
+        /// <param name="samplingKeyColumn">Optional name of the column to use as a stratification column. If two examples share the same value of the <paramref name="samplingKeyColumn"/>
         /// (if provided), they are guaranteed to appear in the same subset (train or test). Use this to make sure there is no label leakage from train to the test set.
         /// If this optional parameter is not provided, a stratification columns will be generated, and its values will be random numbers .</param>
-        /// <param name="seed">Optional parameter used in combination with the <paramref name="stratificationColumn"/>.
-        /// If the <paramref name="stratificationColumn"/> is not provided, the random numbers generated to create it, will use this seed as value.
+        /// <param name="seed">Optional parameter used in combination with the <paramref name="samplingKeyColumn"/>.
+        /// If the <paramref name="samplingKeyColumn"/> is not provided, the random numbers generated to create it, will use this seed as value.
         /// And if it is not provided, the default value will be used.</param>
         /// <returns>Per-fold results: metrics, models, scored datasets.</returns>
         public CrossValidationResult<RegressionMetrics>[] CrossValidate(
-            IDataView data, IEstimator<ITransformer> estimator, int numFolds = 5, string labelColumn = DefaultColumnNames.Label,
-            string stratificationColumn = null, int? seed = null)
+            IDataView data, IEstimator<ITransformer> estimator, int numberOfFolds = 5, string labelColumnName = DefaultColumnNames.Label,
+            string samplingKeyColumn = null, int? seed = null)
```

Comment:
>samplingKeyColumn [](start = 19, length = 17)

Would you consider FoldColumnName?

Reply:
Wei-Sheng suggested partitionColumnName`, which I think reflects the role of the column better. 

---
In reply to: [266177816](https://github.com/dotnet/machinelearning/pull/2973#discussion_r266177816) [](ancestors = 266177816)

Reply:
https://github.com/dotnet/machinelearning/pull/2537
https://github.com/dotnet/machinelearning/issues/2536
We had discussion regarding name of this column.
If you want to change it again, I would advice ask opinion of @rogancarr  and @TomFinley

Reply:
So, I also considered in the issues @Ivanidzo4ka linked the merits of "partition" in my own thinking, though I did not write it down (as I had internally rejected it). I will now write down why I rejected it *here*, and you can see if you agree or disagree.

The major objection is that, if I look forward, I anticipate a future where this same sort of key is useful in *any* sampling utilities. (There are multiple ways to sample a dataset than simple partitionings!) For the sake of that more forward looking perspective, I wanted it named something else. And we ultimately settled on "sampling key."

So I would prefer if we did not act on this suggestion.

Reply:
Incidentally I wouldn't consider it a disaster if we did this. (Keeping it as stratification would have been a disaster! 😄) It merely is slightly misleading I think. But of course I also see @wschin's point.

Reply:
Thank you @Ivanidzo4ka, and @TomFinley for  your comments on this, I am reverting the commit to leave the name of `samplingKeyColumnName` as is.